source/mge.js

```diff
@@ -0,0 +1,988 @@
+
+// Experimental
+
+var megengine = megengine || {};
+var python = python || require('./python');
+var base = base || require('./base');
+
+megengine.ModelFactory = class {
+
+    match(context) {
+        return megengine.Container.open(context);
+    }
+
+    open(context, match) {
+        const identifier = context.identifier;
+        return megengine.Metadata.open(context).then((metadata) => {
```

Comment:
Replace with default implementation? Why is a custom `Metadata` implementation needed?

`return context.metadata('megengine-metadata.json').then((metadata) => {`

Reply:
done